server.py

```diff
@@ -23,53 +28,137 @@
 def hello():
     return 'hello!'
 
-# Endpoint for PUT method
-@app.route('/write', methods=['PUT'])
+@app.route('/redirect')
+def redirect_endpoint():
+    return redirect('http://localhost:5000/read?uuid=xxx', code=302)
+
+# Endpoint for write method
+@app.route('/write', methods=['POST'])
 def write_file():
     file = request.files['file']
     if file:
         filename = secure_filename(file.filename)
         file_uuid = str(uuid.uuid4())
-        # return str(os.path.join(app.config['UPLOAD_FOLDER'], file_uuid)), 200
         file.save(os.path.join(app.config['UPLOAD_FOLDER'], file_uuid))
+        metadata = metadata_manager.MetadataManager()
```

Comment:
I was thinking about this but I got weird errors about threads, so I did it this way.